src/urllib3/util/retry.py

```diff
@@ -353,38 +368,41 @@ def is_retry(self, method, status_code, has_retry_after=False):
         if self.status_forcelist and status_code in self.status_forcelist:
             return True
 
-        return (
+        return bool(
             self.total
             and self.respect_retry_after_header
             and has_retry_after
             and (status_code in self.RETRY_AFTER_STATUS_CODES)
         )
 
-    def is_exhausted(self):
+    def is_exhausted(self) -> bool:
         """ Are we out of retries? """
-        retry_counts = (
-            self.total,
-            self.connect,
-            self.read,
-            self.redirect,
-            self.status,
-            self.other,
-        )
-        retry_counts = list(filter(None, retry_counts))
+        retry_counts = [
+            x
+            for x in (
+                self.total,
+                self.connect,
+                self.read,
+                self.redirect,
+                self.status,
+                self.other,
+            )
+            if x
```

Comment:
We need to do `if x is not None` otherwise we're excluding `0`

Reply:
I think excluding `0` was the behavior:

```
>>> a = (0,)
>>> [x for x in a if x is not None]
[0]
>>> list(filter(None, a))
[]
>>> [x for x in a if x]
[]
```

Also, I did it but I saw that we lost coverage for [this line](https://github.com/urllib3/urllib3/blob/04bf28eb40c68735c719ae044b0af00c23186b03/src/urllib3/util/retry.py#L375).